### ISSUE TITLE
Screen readers may miss inline errors on form controls

### ISSUE THREAD

**Issue by gnomeontherun** [vmware/clarity#3556]
## Describe the bug

Screen reader users may miss the inline errors. While the inline errors are readable using the reading cursor, screen reader users do not hear a status update telling them that the application is in an error state. Therefore, they will not know to look for the inline error messages. Furthermore, since the inline errors are not programmatically associated with their respective fields, screen reader users will not hear the errors when they read the fields as expected while the form is an an error state.

## How to reproduce

https://stackblitz.com/github/clane/clarityTest2/tree/master/clarity-test-app

- Use screen reader
- Activate the Form Demos link
- Press the Tab key to focus the text field in the 'Helper and validation messages' example
- Press the Tab key again or press Shift + Tab
- Observe that you do not hear the following - 'Error message that appears after focus is lost and control is invalid'

Using VoiceOver's reading cursor

- Refresh the page
- Read the text field in the 'Helper and validation messages' example using Ctrl + Shift + left or right arrow key
- Move the reading cursor away from the text field using Ctrl + Shift + left or right arrow key
- Observe that you do not hear the following - 'Error message that appears after focus is lost and control is invalid'

## Expected behavior

_A clear and concise description of what you expected to happen._

## Versions

**App**

* Angular: [e.g. 6]
* Node: [e.g. 8.10.0]
* Clarity: [e.g. 0.12.5]

**Device:**

* Type: [e.g. MacBook]
* OS: [e.g. iOS]
* Browser [e.g. chrome, safari]
* Version [e.g. 22]

## Additional notes
- [x] Use aria-describedby (https://www.w3.org/TR/wai-aria-1.1/#aria-describedby) to enable screen readers to announce error specific information when reading the fields in error by programmatically associating the errors with their respective fields. 
- [ ] Use an assertive live region to announce a status message for the entire form like "There are errors with the following fields: field 1, field 2 ..."
- [ ] If there are multiple errors, focus should be moved to first field in error

**Comment by coryrylan** [comment]
The first task I believe is already complete from earlier issues being closed.

The auto focus to first invalid item should have a way to disable. Large applications that have nested or reusable forms it is not uncommon for the input to not be visible (multi step forms) and auto jumping on submit could be a breaking issue for some forms.

**Comment by coryrylan** [comment]
https://stackblitz.com/edit/angular-fesawc?file=src%2Fapp%2Fform.directive.ts
